modP2pImpl/test/org/aion/p2p/impl/comm/NodeMgrTest.java

```diff
@@ -292,89 +335,400 @@ public void test_getActiveNodesList() {
+
+        node = nMgr.allocNode(ip1, 0);
+        addNodetoOutbound(node, UUID.randomUUID());
+
+        nMgr.movePeerToActive(node.getIdHash(), "outbound");
+
+        node = nMgr.allocNode("1.1.1.1", 0);
+        addNodetoInbound(node, UUID.randomUUID());
+
+        assertEquals(1, nMgr.activeNodesSize());
+
+        nMgr.shutdown();
+        assertTrue(nMgr.getActiveNodesMap().isEmpty());
     }
 
+    @Test
+    public void testDumpNodeInfo() {
+        String dump = nMgr.dumpNodeInfo("testId", false);
+        assertNotNull(dump);
+
+        INode node = nMgr.allocNode(ip1, 0);
+        addNodetoOutbound(node, UUID.randomUUID());
+        nMgr.movePeerToActive(node.getIdHash(), "outbound");
+
+        String dump2 = nMgr.dumpNodeInfo("testId", false);
+        assertEquals(dump.length(), dump2.length());
+
+        String dump3 = nMgr.dumpNodeInfo("testId", true);
+        assertTrue(dump3.length() > dump2.length());
+    }
+
+    @Test
+    public void testDumpNodeInfo2() {
+        String dump = nMgr.dumpNodeInfo("testId", false);
+        assertNotNull(dump);
+
+        nMgr.seedIpAdd(ip1);
+        INode node = nMgr.allocNode(ip1, 0);
+        byte[] rHash = new byte[32];
+        r.nextBytes(rHash);
+        node.updateStatus(r.nextLong(), rHash, BigInteger.ONE);
+        addNodetoOutbound(node, UUID.randomUUID());
+        nMgr.movePeerToActive(node.getIdHash(), "outbound");
+
+        String dump2 = nMgr.dumpNodeInfo("testId", false);
+        assertTrue(dump2.length() > dump.length());
+
+        String dump3 = nMgr.dumpNodeInfo("testId", true);
+        assertEquals(dump3.length(), dump2.length());
+    }
+
+    @Test
+    public void testDumpNodeInfo3() {
+        String dump = nMgr.dumpNodeInfo("testId", false);
+        assertNotNull(dump);
+        addActiveNode();
+        addActiveNode();
+        addActiveNode();
+
+        String dump2 = nMgr.dumpNodeInfo("testId", true);
+        LOGGER.info(dump2);
+        assertTrue(dump2.length() > dump.length());
+    }
+
+    @Test
+    public void testConcurrency() throws InterruptedException {
+        AtomicInteger count = new AtomicInteger(1000);
+
+        AtomicBoolean start = new AtomicBoolean(false);
+
+        BlockingQueue inbound = new LinkedBlockingQueue<INode>();
+        BlockingQueue outbound = new LinkedBlockingQueue<INode>();
+
+        Thread tGenTempNode = new Thread() {
+            @Override
+            public void run() {
+                while (start.get()) {
+                    nMgr.addTempNode(genNode());
+                    try {
+                        Thread.sleep(r.nextInt(5) + 5);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+
+            private INode genNode() {
+                INode n = new Node(false, UUID.randomUUID().toString().getBytes(),
+                    randomIP().getBytes(), r.nextInt(65535) + 1);
```

Comment:
Not sure if the trace logs will already show the IP, but if not can you print it out (for analyzing errors)?

Reply:
fixed in b3bb7020e8e8ff5c13aa516d6c1122a4894ed03d